### PR TITLE
export cmake cyber

### PR DESCRIPTION
**Pull request by ytiang**
merge children libraries into a large library.

### CODE REVIEW

File: src/apollo/cyber/proto/CMakeLists.txt [hunk 1]
```diff
@@ -19,10 +19,11 @@ set(PROTOS choreography_conf.proto
 
 PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS ${PROTOS})
 
-include_directories(${PROTOBUF_INCLUDE_DIRS})
 
 add_library(cyber_proto ${PROTO_SRCS} ${PROTO_HDRS})
 target_link_libraries(cyber_proto ${PROTOBUF_LIBRARIES})
+target_include_directories(cyber_proto PUBLIC ${PROTOBUF_INCLUDE_DIRS}
+  INTERFACE ${CMAKE_INSTALL_PREFIX})
```
Comment (yuzhangbit) [inline src/apollo/cyber/proto/CMakeLists.txt]: why `INTERFACE` ? 
Use generator Expressions for this?
```
$<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}>
```

Comment (ytiang) [reply src/apollo/cyber/proto/CMakeLists.txt]: Key word `INTERFAE` specifies the installation path, So I think `INTERFACE` is equal to `$<INSTALL_INTERFACE:xxx>`. When the key word is `PUBLIC`, generator expressions are often used to specify building paths and installation paths respectively. Such as `$<BUILDING_INTERFACE:xxx>` and `$<INSTALL_INTERFACE:xxx>`

Comment (yuzhangbit) [reply src/apollo/cyber/proto/CMakeLists.txt]: Yes. They are equal in terms of effects. The reason you add INTERFACE is that you want users to find these header files in config mode, right? This is clearly a usage requirement. These headers are also public. So using  `$<INSTALL_INTERFACE:xxx>` is more clear.

File: src/apollo/cyber/CMakeLists.txt [hunk 2]
```diff
@@ -2,16 +2,19 @@ cmake_minimum_required(VERSION 3.5.1)
 
 project(cyber)
 
+set(CMAKE_CXX_STANDARD 14)
```
Comment (yuzhangbit) [inline src/apollo/cyber/CMakeLists.txt]: set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)

File: src/apollo/cyber/croutine/CMakeLists.txt [hunk 3]
```diff
@@ -5,7 +5,7 @@ target_link_libraries(cyber_routine_context cyber::common)
 add_library(cyber::routine_context ALIAS cyber_routine_context)
 
 add_library(cyber_routine_factory routine_factory.h)
-set_target_properties(cyber_routine_factory PROPERTIES LINKER_LANGUAGE CXX)
+set_target_properties(cyber_routine_factory PROPERTIES LINKER_LANGUAGE CXX CXX_STANDARD 14)
```
Comment (yuzhangbit) [inline src/apollo/cyber/croutine/CMakeLists.txt]:
```cmake
set_target_properties(cyber_routine_factory PROPERTIES
    LINKER_LANGUAGE CXX
    CXX_STANDARD 14
    CXX_STANDARD_REQUIRED YES)
target_compile_features(cyber_routine_factory
    PUBLIC
    cxx_lambda_init_captures
    cxx_lambdas)
```

File: src/apollo/cyber/croutine/CMakeLists.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ target_link_libraries(cyber_routine_context cyber::common)
...
 target_link_libraries(cyber_routine_context cyber::common
```
Comment (yuzhangbit) [inline src/apollo/cyber/croutine/CMakeLists.txt]: typo,  should be `cyber_routine_factory`

**Review by yuzhangbit — CHANGES_REQUESTED** [review]

**Review by yuzhangbit — APPROVED** [review]

**Pull request merged**